lib/reline/line_editor.rb

```diff
@@ -2318,23 +1748,23 @@ def finish
     end
     return if h_pointer.nil?
     @history_pointer = h_pointer
+    cursor = current_byte_pointer_cursor
     if @is_multiline
       @buffer_of_lines = Reline::HISTORY[@history_pointer].split("\n")
       @buffer_of_lines = [String.new(encoding: @encoding)] if @buffer_of_lines.empty?
       @line_index = line_no
-      @line = @buffer_of_lines[@line_index]
-      @rerender_all = true
+      calculate_nearest_cursor(cursor)
     else
-      @line = Reline::HISTORY[@history_pointer]
+      @buffer_of_lines = [Reline::HISTORY[@history_pointer]]
+      calculate_nearest_cursor(cursor)
```

Comment:
For my own learning: Why is it necessary to calculate the cursor position here?

Reply:
When `@buffer_of_lines[@line_index]` or `@buffer_of_lines` itself is directly modified, we need set byte_pointer to `0` or `line.bytesize`, or re-calculate byte_position based on previous cursor position to avoid byte_position being in a middle of grapheme cluster.

For this case, I just found a bug like this which is fixed by this calculate_nearest_cursor. 
```
# prepare history
irb(main):001> "1️⃣"
=> "1️⃣"
irb(main):002> "1
# press PGUP and 2
irb(main):002> "12️⃣"
# press DELETE DELETE
irb(main):002> "(garbled text)"
```